GeeksCoreLibrary/Modules/Databases/Services/DocumentStoreConnection.cs

```diff
@@ -0,0 +1,270 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+using DocumentFormat.OpenXml.Spreadsheet;
+using GeeksCoreLibrary.Core.DependencyInjection.Interfaces;
+using GeeksCoreLibrary.Core.Helpers;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using GeeksCoreLibrary.Modules.Databases.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using MySqlX.XDevAPI;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace GeeksCoreLibrary.Modules.Databases.Services;
+
+public class DocumentStoreConnection : IDocumentStoreConnection, IScopedService
+{
+    private readonly IHttpContextAccessor httpContextAccessor;
+    private readonly ILogger<DocumentStoreConnection> logger;
+
+    private Session Session { get; set; }
+
+    private readonly GclSettings gclSettings;
+
+    private readonly ConcurrentDictionary<string, object> parameters = new();
+    private readonly JsonSerializerSettings jsonSerializerSettings;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DocumentStoreConnection"/>.
+    /// </summary>
+    public DocumentStoreConnection(IOptions<GclSettings> gclSettings, IHttpContextAccessor httpContextAccessor, ILogger<DocumentStoreConnection> logger)
+    {
+        this.httpContextAccessor = httpContextAccessor;
+        this.logger = logger;
+        this.gclSettings = gclSettings.Value;
+
+        jsonSerializerSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new DefaultContractResolver
+            {
+                NamingStrategy = new CamelCaseNamingStrategy()
+            },
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+    }
+
+    /// <inheritdoc />
+    public string ConnectedDatabase { get; private set; }
+
+    /// <inheritdoc />
+    public Task CreateCollectionAsync(string collectionName, List<(string Name, DocumentStoreIndexModel Indexes)> collectionIndexes)
+    {
+        EnsureOpenSession();
+
+        var collection = Session.Schema.CreateCollection(collectionName, true);
+        if (collectionIndexes == null)
+        {
+            return Task.CompletedTask;
+        }
+        
+        // Create indices
+        foreach (var (name, indexes) in collectionIndexes)
+        {
+            collection.CreateIndex(name, JsonConvert.SerializeObject(indexes, jsonSerializerSettings));
+        }
+
+        return Task.CompletedTask;
+    }
+    
+    /// <inheritdoc />
+    public Task<bool> CollectionExists(string name)
+    {
+        EnsureOpenSession();
+        var collection = Session.Schema.GetCollection(name);
+
+        return Task.FromResult(collection.ExistsInDatabase());
```

Comment:
Why is `Task.FromResult` used here? It's not an async function. Is it not possible to just return the result of `collection.ExistsInDatabase()`? If it is possible to use it without `Task.FromResult`, please change the return type to `bool` instead of `Task<bool>` in the interface and this class.

Reply:
As discussed: Method returns a task for futureproofing, so we can change implementations to an async variant without it being a breaking change.